File: nomad-samples/python/account_updates/account/change_password.py
from constants.project_constants import *
from exceptions.api_exception_handler import *

import json, requests

def change_password(AUTH_TOKEN: str, CURRENT_PASSWORD: str, NEW_PASSWORD: str) -> dict:
    if (not AUTH_TOKEN):
        raise Exception("Authentication Token: The authentication token is invalid")
  
    API_URL = PORTAL_API_URL + "/account/change-password"
  
    # Create header for the request
    HEADERS = {
        "Content-Type": "application/json",
        "Authorization": "Bearer " + AUTH_TOKEN
    }

    # Build the payload body
    BODY = {
        "password": CURRENT_PASSWORD,
        "newPassword": NEW_PASSWORD
    }
    
    try:
        RESPONSE = requests.post(API_URL, headers=HEADERS, data=json.dumps(BODY))
        
        if not RESPONSE.ok:
            raise Exception()

    except:
        api_exception_handler(RESPONSE, "Change password failed")

File: nomad-samples/portal/js/account-updates/account/change-password.js
import * as prjConstants from "../constants/project-constants.js";
import apiExceptionHandler from "../exceptions/api-exception-handler.js";

export default async function changePassword(AUTH_TOKEN)
{
    // Create header for the request
    const HEADERS = new Headers();
    HEADERS.append("Content-Type", "application/json");
    HEADERS.append("Authorization", `Bearer ${AUTH_TOKEN}`);
  
  	const BODY = {
        password: "currentPassword",
        newPassword: "newPassword"
    };
  
    // Post
    const RESPONSE = await fetch(`${prjConstants.PORTAL_API_URL}/account/change-password`, {
        method: "POST",
        headers: HEADERS,
        body: JSON.stringify(BODY)
    }).catch((exception) => {
        throw exception;
    });

    // Check for success
    if (RESPONSE.ok) {
        return(RESPONSE.status);
    }
		
  	// There was an error
    await apiExceptionHandler(RESPONSE, "Change password failed");
}